src/modules/universal-discovery/universal.discovery.module.js

```diff
@@ -221,25 +245,37 @@ export default class UniversalDiscoveryModule extends Component {
      * @memberof UniversalDiscoveryModule
      */
     checkIsBookmarked(locationId) {
+        if (this.isBookmarkLoading(locationId)) {
```

Comment:
I think you should update state, then re-render a bookmark with loading state.

Reply:
Currently when bookmark is loading there is no bookmark being displayed to user.
The problem is that render function should be pure and you should not update state during rendering, otherwise React shows warnings. This is the reason, why I decided to store ids of loading bookmarks in: `this.loadingBookmarksLocationsIds = {};`.